src/core/state/favorites/index.ts

```diff
@@ -37,32 +38,67 @@ export interface FavoritesState {
 }
 
 export const favoritesStore = createStore<FavoritesState>(
-  (set, get) => ({
-    favorites: getInitialFavorites(),
-    setFavorites: (favorites) => set({ favorites }),
-    addFavorite: ({ address, chainId }: UpdateFavoritesArgs) => {
-      const { favorites } = get();
-      const currentFavorites = favorites[chainId] || [];
-      set({
-        favorites: {
-          ...favorites,
-          [chainId]: [...currentFavorites, address],
+  (set, get) => {
+    // Helper function for tracking analytics
+    const trackFavoriteChange = (
+      address: AddressOrEth,
+      chainId: ChainId,
+      isAdding: boolean,
+    ) => {
+      const updatedState = get();
+      const updatedFavorites = updatedState.favorites[chainId] || [];
+      const afterLength = updatedFavorites.length;
+
+      const analyticsData = {
+        token: {
+          address,
+          chainId,
```

Comment:
Since we're getting favorites and safely accessing favorites by chainId within `addFavorite` and `removeFavorite`, we are kind of duplicating that logic here.

If you pass the edited favorites array (or its length) to this function we could reduce some repetition and then would also be able to remove this from the store's setup completely.